experimental/packages/opentelemetry-sdk-metrics-base/src/state/MetricCollector.ts

```diff
@@ -35,8 +35,9 @@ export class MetricCollector implements MetricProducer {
 
   async collect(): Promise<ResourceMetrics> {
     const collectionTime = hrTime();
-    const instrumentationLibraryMetrics = (await Promise.all(this._sharedState.meterSharedStates
-      .map(meterSharedState => meterSharedState.collect(this, collectionTime))));
+    const meterCollectionPromises = Array.from(this._sharedState.meterSharedStates.values())
+      .map(meterSharedState => meterSharedState.collect(this, collectionTime));
```

Comment:
Random trivia: `Array.from(states.values(), state => state.collect(this, collectionTime));` would be shorter, but benched it for fun and found that `Array.from(states.values()).map(state => state.collect(this, collectionTime));` is ~4x faster on Node.js :man_shrugging: